File: 2785-sort-vowels-in-a-string/2785-sort-vowels-in-a-string.cpp
class Solution {
public:
    bool isVowel (char c) {
        if (
            c == 'A' ||
            c == 'E' ||
            c == 'I' ||
            c == 'O' ||
            c == 'U' ||
            c == 'a' ||
            c == 'e' ||
            c == 'i' ||
            c == 'o' ||
            c == 'u'
        ) return true;

        return false;
    }

    string sortVowels(string s) {
        string vowels = "";

        for (char c : s){
            if (isVowel(c)) {
                vowels += c;
            }
        }

        sort(vowels.begin(), vowels.end());

        int i = 0;
        int j = 0;
    
        while (i < s.size()){
            if (isVowel(s[i])) {
                swap (s[i], vowels[j++]);
            }
            i++;
        }

        return s;
    }
};

File: 2785-sort-vowels-in-a-string/2785-sort-vowels-in-a-string.java
class Solution {
    public boolean isVowel (Character c) {
        if (
            c == 'A' ||
            c == 'E' ||
            c == 'I' ||
            c == 'O' ||
            c == 'U' ||
            c == 'a' ||
            c == 'e' ||
            c == 'i' ||
            c == 'o' ||
            c == 'u'
        ) return true;

        return false;
    }

    public String sortVowels(String s) {
        String vowels = "";

        for (int i=0; i<s.length(); i++){
            if (isVowel (s.charAt(i))) {
                vowels += s.charAt(i);
            }
        }

        char[] v = vowels.toCharArray();
        Arrays.sort(v);

        char[] arr = s.toCharArray();

        int i=0;
        int j=0;

        while (i < arr.length) {
            if (isVowel(arr[i])) {
                arr[i] = v[j++];
            }
            i++;
        }

        return new String(arr);

    }
}